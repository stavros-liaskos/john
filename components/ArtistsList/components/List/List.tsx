import LastFm from '../../../Icons/lastfm';
import Spotify from '../../../Icons/spotify';
import Button from '../../../Button/Button';
import React from 'react';
import { ArtistsListProp } from '../../ArtistsList';

const ICON_SIZE = 30;

const List = ({ i18n, artistsList, onButtonClick, artistLoading }: ArtistsListProp) => {
  if (!i18n || !i18n.btnTxt || !i18n.noArtists || !artistsList?.length || !onButtonClick) {
    return null;
  }

  return (
    <>
      {artistsList.map((artist, index: number) => (
        <div
          className="flex justify-between md:justify-center items-center dark:even:bg-gh-darkly even:bg-gray-100"
          key={artist.id}
        >
          <p className="grow text-clip rr-text">{artist.name}</p>
          <div className="flex basis-2 mx-4 md:mx-8">
            {artist.lastfmUri && (
              <a className="inline" href={artist.lastfmUri}>
                <LastFm width={ICON_SIZE} height={ICON_SIZE} />
              </a>
            )}
            {artist.spotifyUri && (
              <a className="inline" href={artist.spotifyUri}>
                <Spotify width={ICON_SIZE} height={ICON_SIZE} />
              </a>
            )}
          </div>
          <Button
            i18n={i18n.btnTxt}
            handleClick={onButtonClick}
            handleClickArg={artist.id}
            className={`btn-small lg:ml-8 my-2 ${index % 2 ? '!border-gh-dark' : ''}`}
            disabled={!!artistLoading && artist.id === artistLoading}
            loading={!!artistLoading && artist.id === artistLoading}
          />
        </div>
      ))}
    </>
  );
};

export default List;
