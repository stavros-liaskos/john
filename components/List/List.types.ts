export interface ListProps {
  i18n: { unfollow: string };
}
