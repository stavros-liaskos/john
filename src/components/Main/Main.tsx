import FollowedArtistList from '../FollowedArtistList/FollowedArtistList';
import React, { Suspense } from 'react';
import Search from '../Search/Search';
import { useUser } from '@auth0/nextjs-auth0/client';
import Login from '../Login/Login';
import Scrapers from '../Scrape/Scrapers';
import Recommendations from '../Recommendations/Recommendations';
import Loading from '../Loading/Loading';
import { searchI18n, followedArtistListI18n, loginI18n, recommendationsI18n } from '../../i18n';

const Main: React.FunctionComponent = () => {
  const { user } = useUser();

  function loginBE() {
    window.location.assign('https://api.releaseraccoon.online/me');
  }

  function fetchy() {
    fetch('https://api.releaseraccoon.online/me', {
      headers: { 'X-Requested-With': 'JavaScript' },
    });
  }

  return (
    <main className="rr-column flex-auto">
      <div className="flex flex-col flex-auto w-full lg:w-9/12">
        <button className={'rr-text btn'} onClick={loginBE}>
          login with assign
        </button>

        <button className={'rr-text btn'} onClick={fetchy}>
          login with fetch
        </button>

        <a href="https://api.releaseraccoon.online/me">
          <button className={'rr-text btn'}>login with anchor</button>
        </a>

        {user ? (
          <>
            <Search i18n={searchI18n} />
            <Scrapers />
            <div className="flex flex-auto flex-col h24">
              <FollowedArtistList i18n={followedArtistListI18n} />
              <Suspense fallback={<Loading />}>
                <Recommendations i18n={recommendationsI18n} />
              </Suspense>
            </div>
          </>
        ) : (
          <Login i18n={loginI18n} />
        )}
      </div>
    </main>
  );
};
Main.whyDidYouRender = false;
export default Main;
