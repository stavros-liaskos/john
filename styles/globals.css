/* purgecss start ignore */
@tailwind base;
@tailwind components;
/* purgecss end ignore */
@tailwind utilities;

@layer components {
  .rr-border {
    @apply dark:border-gh-border border-black;
  }
  .rr-text {
    @apply dark:text-slate-300;
  }
  .h3 {
    @apply rr-text mb-4;
  }
  .btn {
    @apply font-semibold text-sm rounded-sm shadow border-2 rr-border rr-text flex items-center justify-center;
  }
  .btn-large {
    @apply py-2 px-3;
  }
  .btn-small {
    @apply py-1 px-3;
  }
}
