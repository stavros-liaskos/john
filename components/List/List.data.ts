import { ListEl } from './List.types';

export const listData: ListEl[] = [
  {
    followedByUser: false,
    lastfmUri: 'https://www.last.fm/music/Liedfett',
    name: 'Liedfett',
    spotifyUri: '',
  },
  {
    followedByUser: false,
    lastfmUri: 'https://www.last.fm/music/Leaders+of+the+New+School',
    name: 'Leaders of the New School',
    spotifyUri: '',
  },
  {
    followedByUser: false,
    lastfmUri: 'https://www.last.fm/music/lo+&+leduc',
    name: 'lo & leduc',
    spotifyUri: '',
  },
];

export const listI18n = {
  unfollow: 'unfollow',
};
