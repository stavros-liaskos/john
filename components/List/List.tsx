import Image from 'next/image';
import { ListProps, ListEl } from './List.types';
import React from 'react';

const List: React.FunctionComponent<ListProps> = ({ list }) => {
  if (!list) {
    return null;
  }

  return (
    <div className="border border-indigo-100">
      {list.map((el: ListEl, index: number) => (
        <div className="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl" key={index}>
          <div className="md:flex">
            <div className="md:shrink-0">
              <Image
                className="h-48 w-full object-cover md:h-full md:w-48"
                src="https://www.placecage.com/120/120"
                alt={`Logo of ${el.name}`}
                width={120}
                height={170}
              />
            </div>
            <div className="p-8">
              <div className="uppercase tracking-wide text-sm text-indigo-500 font-semibold">Case study</div>
              <a href="#" className="block mt-1 text-lg leading-tight font-medium text-black hover:underline">
                {el.name}
              </a>
              <p className="mt-2 text-gray-500">
                Getting a new business off the ground is a lot of hard work. Here are five ideas you can use to find
                your first customers.
              </p>
            </div>
          </div>
        </div>
      ))}
    </div>
  );
};

export default List;
