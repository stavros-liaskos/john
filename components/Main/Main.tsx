import { MainProps } from './Main.types';
import List from '../List/List';
import React, { useState } from 'react';
import { ListEl } from '../List/List.types';
import Search from '../Search/Search';
import { searchI18n } from '../Search/Search.data';

const list2 = [
  {
    name: 'Led Zeppeling',
    href: 'www.google.com',
  },
  {
    name: 'YOYOO',
    href: 'www.google.com',
  },
];

const Main: React.FunctionComponent<MainProps> = ({ i18n, className, defaultList = [] }) => {
  const [list, setList] = useState<ListEl[]>(defaultList);

  if (!i18n || !i18n.todo) {
    return null;
  }

  const handleSearch = (query: string) => {
    fetch('https://jsonplaceholder.typicode.com/todos', {
      method: 'POST',
      body: new URLSearchParams(`query=${query}`),
    })
      .then(res => res.json())
      .then(result => setList(list2)) // TODO replace list2 with result
      .catch(() => {});
  };

  return (
    <main className={`${className}`}>
      <Search i18n={searchI18n} handleSearch={handleSearch} />

      <List list={list} />
    </main>
  );
};

export default Main;
