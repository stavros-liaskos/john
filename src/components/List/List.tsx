import React from 'react';
import { ListProps, ListEl } from './List.types';
import Spotify from '../Icons/spotify';
import LastFm from '../Icons/lastfm';

const List: React.FunctionComponent<ListProps> = ({ list, i18n }) => {
  if (!list || !i18n || !i18n.unfollow) {
    return null;
  }

  return (
    <div className="border border-indigo-100">
      {list.map((el: ListEl, index: number) => (
        <div className="flex justify-between items-center px-8 even:bg-gray-100" key={index}>
          <p>{el.name}</p>
          <div className="flex">
            {el.lastFmUrl && (
              <a className="inline" href={el.lastFmUrl}>
                <LastFm />
              </a>
            )}
            {el.spotifyUrl && (
              <a className="inline" href={el.spotifyUrl}>
                <Spotify />
              </a>
            )}
          </div>
          <button className="py-2 px-3 bg-indigo-800 text-white text-sm font-semibold rounded-md shadow focus:outline-none">
            {i18n.unfollow}
          </button>
        </div>
      ))}
    </div>
  );
};

export default List;
