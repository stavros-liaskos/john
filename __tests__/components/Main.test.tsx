import React from 'react';
import { render, getByRole, queryAllByRole } from '@testing-library/react';
import Main from '../../components/Main/Main';
import { mainI18n } from '../../components/Main/Main.data';
import { expect } from '@jest/globals';
import { listData } from '../../components/List/List.data';
import { MainProps } from '../../components/Main/Main.types';

const setup = (props?: Partial<MainProps>) => {
  const { container } = render(<Main className="" i18n={mainI18n} {...props} />);
  const search = getByRole(container, 'button');
  const artistLogos = queryAllByRole(container, 'img');
  const input = getByRole(container, 'textbox');

  return { container, search, artistLogos, input };
};

describe('Main', () => {
  it('renders without data without crashing', () => {
    // @ts-ignore
    render(<Main />);
  });

  it('show only Search without list prop', () => {
    const { container, search, artistLogos } = setup();

    expect(search).toBeTruthy();
    expect(artistLogos.length).toEqual(0);
    expect(container).toMatchSnapshot();
  });

  it('show List of results ', async () => {
    const { container, search, artistLogos } = setup({
      defaultList: listData,
    });

    expect(search).toBeTruthy();
    expect(artistLogos.length).toEqual(2);
    expect(container).toMatchSnapshot();
  });
});
