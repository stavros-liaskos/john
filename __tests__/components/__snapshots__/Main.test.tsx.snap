// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main show List of results  1`] = `
<div>
  <main
    class=""
  >
    <div
      class="flex justify-center items-center h-48 bg-red-300"
    >
      <form
        class="flex justify-center items-stretch h-10"
        novalidate=""
      >
        <input
          class="rounded mr-4 px-2 max-w-md"
          name="search"
          placeholder="Here goes the artist..."
          type="text"
        />
        <button
          class="py-2 px-3 bg-indigo-800 text-white text-sm font-semibold rounded-md shadow focus:outline-none"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="border border-indigo-100"
    >
      <div
        class="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl"
      >
        <div
          class="md:flex"
        >
          <div
            class="md:shrink-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTIwIiBoZWlnaHQ9IjE3MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logo of Led Zeppeling"
                class="h-48 w-full object-cover md:h-full md:w-48"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="p-8"
          >
            <div
              class="uppercase tracking-wide text-sm text-indigo-500 font-semibold"
            >
              Case study
            </div>
            <a
              class="block mt-1 text-lg leading-tight font-medium text-black hover:underline"
              href="#"
            >
              Led Zeppeling
            </a>
            <p
              class="mt-2 text-gray-500"
            >
              Getting a new business off the ground is a lot of hard work. Here are five ideas you can use to find your first customers.
            </p>
          </div>
        </div>
      </div>
      <div
        class="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl"
      >
        <div
          class="md:flex"
        >
          <div
            class="md:shrink-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTIwIiBoZWlnaHQ9IjE3MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logo of Led Zeppeling"
                class="h-48 w-full object-cover md:h-full md:w-48"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="p-8"
          >
            <div
              class="uppercase tracking-wide text-sm text-indigo-500 font-semibold"
            >
              Case study
            </div>
            <a
              class="block mt-1 text-lg leading-tight font-medium text-black hover:underline"
              href="#"
            >
              Led Zeppeling
            </a>
            <p
              class="mt-2 text-gray-500"
            >
              Getting a new business off the ground is a lot of hard work. Here are five ideas you can use to find your first customers.
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Main show only Search without list prop 1`] = `
<div>
  <main
    class=""
  >
    <div
      class="flex justify-center items-center h-48 bg-red-300"
    >
      <form
        class="flex justify-center items-stretch h-10"
        novalidate=""
      >
        <input
          class="rounded mr-4 px-2 max-w-md"
          name="search"
          placeholder="Here goes the artist..."
          type="text"
        />
        <button
          class="py-2 px-3 bg-indigo-800 text-white text-sm font-semibold rounded-md shadow focus:outline-none"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="border border-indigo-100"
    />
  </main>
</div>
`;
